src/pages/newsDetail.tsx

```diff
@@ -2,16 +2,18 @@ import React, { FunctionComponent } from "react";
 import { ChevronLeft } from "react-feather";
 import Link from "next/link";
 import { Helmet } from "react-helmet";
-import { useLocation, useParams } from "react-router-dom";
 import { allNews } from "./../components/News";
 import { NewsSingle, NewsTag } from "./../components/News/types";
 import { NewsDetailContent } from "./../components/News/NewsDetailContent/NewsDetailContent";
 import { ResourceDownload } from "./../components/UI/ResourceDownload";
+import { useRouter } from "next/router";
 
 export const NewsPageDetail: FunctionComponent = () => {
-  const location = useLocation();
-  const params: { slug: string } = useParams();
-  const detail: NewsSingle = allNews.find((news) => news.slug === params.slug);
+  const router = useRouter();
+
+  const slug = router.pathname.split("news/")[1];
```

Comment:
https://nextjs.org/docs/routing/dynamic-routes

should be retrievable without the need to split?

Reply:
yeap, but i'm trying to refactor bit by bit so this is to allow me to remove react router dom. Will get the actual slug when i actually move the page.